wrap/python/pybind11/.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "tools/clang"]
```

Comment:
OK. if pybind itself has submodules, should we try to do pybind as well with a submodule? What are the pros and the cons?

Reply:
Pros:
- Less diff when we update the module

Cons:

- Github Release cannot produce a tarball with the module included (I encountered this yesterday with another repo, the tarball does not build because of a missing module). See https://github.community/t5/How-to-use-Git-and-GitHub/Create-Release-that-contains-submodule/td-p/12073
- People need to do `git clone --recursive` or they will have a error message that pybind11 is not found

I think I can separate the pybind11 folder addition into another PR.